subsys/net/openthread/rpc/client/ot_rpc_commissioning_client.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <ot_rpc_ids.h>
+#include <ot_rpc_types.h>
+#include <ot_rpc_common.h>
+
+#include <nrf_rpc/nrf_rpc_serialize.h>
+#include <nrf_rpc/nrf_rpc_cbkproxy.h>
+
+static void ot_rpc_thread_discover_cb_rpc_handler(const struct nrf_rpc_group *group,
+						  struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	otHandleActiveScanResult callback_slot;
+	otActiveScanResult aResult;
+	struct ot_rpc_active_scan_result result;
+	void *pkt_data = &result;
+	void *context;
+
+	pkt_data = nrf_rpc_decode_buffer(ctx, pkt_data, sizeof(result));
+	if (pkt_data) {
+		memcpy(aResult.mExtAddress.m8, result.ext_addr, OT_EXT_ADDRESS_SIZE);
+		memcpy(aResult.mNetworkName.m8, result.nwk_name, OT_NETWORK_NAME_MAX_SIZE + 1);
+		memcpy(aResult.mExtendedPanId.m8, result.ext_pan_id, OT_EXT_PAN_ID_SIZE);
+		memcpy(aResult.mSteeringData.m8, result.steering_data, OT_STEERING_DATA_MAX_LENGTH);
+		aResult.mPanId = result.pan_id;
```

Comment:
Hmm. This assumes `pan_id` is aligned while `result` is a packed structure, so it may not be, right? Probably using some Zephyr helpers for copying int byte by byte might be more portable/future-proof.

Reply:
It was redesigned, not actual anymore.
Nevertheless, this implementation is correct. Compiler follows up data structure attributes and generates alignment safe backend code to get access to packed members.
The nonalignment fault is possible only if the type was cast explicitly and the compiler lost knowledge about packing in a new type.

Reply:
btw. this is classic approach for parsing \ composing of serialized data without taking into account shifts, positioning etc. Compiler will do everything for you by self on safe manner.
Read it in some computer science book, unfortunately forgot which one. :)

Reply:
Ok, never found any GCC guarantees that this is handled already, but I checked in assembly that you're right. Thanks!

Reply:
Yep, I checked it too on previous projects where I participated with several compilers and platforms. GCC, IAR, Keil all are managed this correctly for different platforms (checked for arm, avr32, arc).  I guess this is considered by any compiler developers.